sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2641,4 +2641,12 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     val idTuples = sampled.collect().map(row => row.getLong(0) -> row.getLong(1))
     assert(idTuples.length == idTuples.toSet.size)
   }
+
+  test("SPARK-30811: CTE should not cause stack overflow when " +
```

Comment:
Can we have a more closer unit test in `AnalysisErrorSuite` instead of here?

Reply:
Sure, I can add one. However I do think that for regressions an end to end test adds value.

Reply:
Got it, @hvanhovell .